### DISCUSSION TITLE
Can import anterior price and autocalculate the discount

### DISCUSSION THREAD

**Discussion by LinkOfLight** [PrestaShop/PrestaShop#40241] [Ideas]
**Is your feature request related to a problem? Please describe.**
I think the import page can import anterior Price and calculate discount in the import

**Describe the solution you'd like**
Can Import the anterior Price whit the import page and auto calculate the discout

**Describe alternatives you've considered**

**Additional context**
A lot of providers make the csv whit the anterior price

![image](https://user-images.githubusercontent.com/9163980/48978280-823d6100-f0a9-11e8-9f08-f18ae27802a5.png)

**Comment by khouloudbelguith** [comment]
Hi @LinkOfLight,

Thanks for your suggestion.
@marionf what do you think, can we add the option "Previous price" into the import page for the Product?
In our case, we have Price tax excluded or Price tax included / Discount amount or Discount percent
![image](https://user-images.githubusercontent.com/16067358/49007823-b29a0380-f16c-11e8-9afd-5a14eee4f296.png)
Thanks!

**Comment by LinkOfLight** [comment]
I know, but our provider gives us the previous price to download the CSV, not the discount, it would be a great improvement to implement that import the previous price and calculate, taking into account the price with VAT included or excluding the discount, thus being able to facilitate to those who We do dropshipping a lot of putting our offers

**Comment by colinegin** [comment]
Hello @LinkOfLight , thanks for submitting your idea ! 
Just to be sure I understood your need : currently you have the basis price of your product and its final price (with the discount), but you don't know what is the value of this discount, right ?

I'm not sure I understand why you need to have the "previous price" (=basis / initial price), as it is supposed to be the Price tax excluded / included field as shown in your template.
What you need would rather be the final price field, right ?

The issue I see is that if we have the basis and the final price, it would need to automatically calculate the  discount amount. But for instance, if the basis price is 80€ and the final price is 60€, how do we know if the discount is 25% or 20€ ?

Let me know what you think !

Have a nice day,

**Comment by LinkOfLight** [comment]
My provider send my the final Price (pvp) whit taxes (and discount aplicated) and the final Price whitout discound and taxes aplicated, i think the import cant calculated the € discount and whit this the % ¿for what? because whit this the sistema can aplicated the "in offert" and can show the two prices, think i have more tan 4000 products and my provider send my all the prices, make the "in offert" is a titanic quest, i need refresh the prices whit csv al the days (sorry for my bad english)

**Comment by colinegin** [comment]
Hello @LinkOfLight , just to be sure I understood what you told me, your provider sends you : 
- the final price that is displayed to the customer (ex : 40€ tax included)
- the basis price without taxes and discount (ex : 50€)
Again, how do you know if in FO, there should be a label saying -10€ discount or -20% discount ?

I'm sorry i'm not sure I understand what you mean with "in offert".

**Comment by LinkOfLight** [comment]
Sorry for my bad explication, My supplier gives me the previous price with taxes and the new price with taxes, the idea is that prestashop calculate with the new and the old the applied discount and can show the label of "on offer" with the label of the discount applied

**Comment by colinegin** [comment]
Alright, I understand ! But as I said last time, how can the software know if you want to show your discount in percentage or in amount ? Ex : if the basis price is 80€ and the final price is 60€, how do we know if the discount is 25% or 20€ ?

Thank you

Coline

**Comment by LinkOfLight** [comment]
i think the 20€ is better but if we can chose the label in % or € is better

**Comment by colinegin** [comment]
I'm not sure to understand the added value if in the end you need to choose the prefered option ?

Anyway we will rework the import page in the next versions, and will review your feedback at that moment.

Thanks !

**Comment by LinkOfLight** [comment]
yep, if we can select the prefered choice is better
